compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -32,10 +32,12 @@ class CheckUnused extends Phase:
 
   override def run(using Context): Unit =
     val tree = ctx.compilationUnit.tpdTree
-    val data = UnusedData()
-    val fresh = ctx.fresh.setProperty(_key, data)
-    traverser.traverse(tree)(using fresh)
-    reportUnusedImport(data.getUnused)
+    val data = UnusedData(ctx)
+    if data.neededChecks.nonEmpty then
```

Comment:
you can override `isRunnable` which won't even run the phase if this is false, rather than running this check on every compilation unit